### PR TITLE
1682: Generate the web showcase

### PR DESCRIPTION
**Pull request by thisAAY**
- Implement a Web template for the showcase as a web page
- Implemented in HTML, CSS, and JS Thanks to ChatGPT 👀
- The web page consists of these functions
    1. List all components
    2. Search by name, package name, group, or tags
    3. Filter by any key
- The template can change in the future but in my opinion, this is enough for the MVP

https://github.com/user-attachments/assets/e280eada-e6ac-4131-9386-7f4e5b1e9d70

### CODE REVIEW

File: plugin/src/main/java/com/infinum/arkive/plugin/generators/ShowcaseWebGenerator.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+}
+
+private const val INDEX_FILE = "index.html"
+private const val STYLES_FILE = "styles.css"
+private const val JS_FILE = "script.js"
+private const val RESOURCES_DIR = "web"
+
+class ShowcaseWebGeneratorImpl : ShowcaseWebGenerator {
+    // Best way to get class loader in a plugin
+    private val classLoader = Thread.currentThread().contextClassLoader
```
Comment (antunflas) [inline plugin/src/main/java/com/infinum/arkive/plugin/generators/ShowcaseWebGenerator.kt:17]: Hmm, interesting.. I was using an object generated in our source code and calling `::class.java.getResourceAsStream`. What are the pros/cons?

File: plugin/src/main/java/com/infinum/arkive/plugin/tasks/GenerateWebShowcaseTask.kt [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import org.gradle.api.tasks.PathSensitive
+import org.gradle.api.tasks.PathSensitivity
+import org.gradle.api.tasks.SkipWhenEmpty
+import org.gradle.api.tasks.TaskAction
+
+@CacheableTask
+internal open class GenerateWebShowcaseTask : BaseSourceTask() {
+
+    @get:OutputDirectory
+    var outputDirectory: File = project.layout.buildDirectory.get().file(
```
Comment (antunflas) [inline plugin/src/main/java/com/infinum/arkive/plugin/tasks/GenerateWebShowcaseTask.kt]: `Property`

Comment (antunflas) [reply plugin/src/main/java/com/infinum/arkive/plugin/tasks/GenerateWebShowcaseTask.kt]: Also, when there are multiple tasks, consider making the Plugin's extension so one does not have to configure each task, but can configure the extension. Especially important if there are configurable values that need to be shared across the tasks.

File: plugin/src/main/resources/web/script.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  const grid = document.getElementById('grid');
+  const searchBar = document.getElementById('search-bar');
+  const filterDropdown = document.getElementById('filter-dropdown');
+
+  let components = [];
+
+  // Load JSON data
+  const loadComponents = async () => {
+    try {
+      const response = await fetch('arkive-showcase.json');
```
Comment (antunflas) [inline plugin/src/main/resources/web/script.js:11]: How is this json file accessible just by the name?

**Review by antunflas — APPROVED** [review]
This looks very nice 👍 
Could you please just add a border (or any other indicator such as an "empty" background) around the snapshot image in html so it's clear what the bounds of the preview are?

File: plugin/src/main/java/com/infinum/arkive/plugin/generators/ShowcaseWebGenerator.kt (hunk 1 above)
Comment (thisAAY) [reply plugin/src/main/java/com/infinum/arkive/plugin/generators/ShowcaseWebGenerator.kt:17]: was it a convention plugin? or binary one? 

Some of class loaders don't work when packaged inside a jar which is the case with this plugin

File: plugin/src/main/resources/web/script.js (hunk 3 above)
Comment (thisAAY) [reply plugin/src/main/resources/web/script.js:11]: All of these files are generated in the same directory, to simplify uploading

also let's huddle next week, want to know how to publish this file to github-pages

File: plugin/src/main/java/com/infinum/arkive/plugin/generators/ShowcaseWebGenerator.kt (hunk 1 above)
Comment (antunflas) [reply plugin/src/main/java/com/infinum/arkive/plugin/generators/ShowcaseWebGenerator.kt:17]: It was a plugin published to mavenLocal as a jar.

Comment (thisAAY) [reply plugin/src/main/java/com/infinum/arkive/plugin/generators/ShowcaseWebGenerator.kt:17]: Just tried it, didn't work, I will stick with this for now

`Thread.currentThread().contextClassLoader`

**Pull request merged**